app/serializers/concerns/topic_tags_mixin.rb

```diff
@@ -11,11 +11,16 @@ def include_tags?
 
   def tags
     # Calling method `pluck` or `order` along with `includes` causing N+1 queries
-    tags = (SiteSetting.tags_sort_alphabetically ? topic.tags.sort_by(&:name) : topic.tags.sort_by(&:topic_count).reverse).map(&:name)
+    tags = (SiteSetting.tags_sort_alphabetically ? topic.tags.sort_by(&:name) : topic.tags.sort_by(&:topic_count).reverse).map do |tag|
+      {
+        name: tag.name,
+        description: tag.description
+      }.compact
```

Comment:
I think `.compact` should be  on the next line?

Reply:
Also I am a little worried this could potentially break some things that are just expecting an array of strings an not an array of hashes...have you taken a look at the places using this?